src/snowflake/snowpark/mock/_stored_procedure.py

```diff
@@ -97,7 +116,56 @@ def __call__(
             else:
                 parsed_args.append(arg)
 
-        result = self.func(session, *parsed_args)
+                # Initialize import directory
+
+        temporary_import_path = tempfile.TemporaryDirectory()
+        last_import_directory = sys._xoptions.get("snowflake_import_directory")
+        sys._xoptions["snowflake_import_directory"] = temporary_import_path.name
+
+        # Save a copy of module cache
+        frozen_sys_module_keys = set(sys.modules.keys())
+
+        def cleanup_imports():
+            for module_path in self.imports:
+                if module_path in sys.path:
+                    sys.path.remove(module_path)
+
+            # Clear added entries in sys.modules cache
+            added_keys = set(sys.modules.keys()) - frozen_sys_module_keys
+            for key in added_keys:
+                del sys.modules[key]
+
+            # Cleanup import directory
+            temporary_import_path.cleanup()
+
+            # Restore snowflake_import_directory
+            if last_import_directory is not None:
+                sys._xoptions["snowflake_import_directory"] = last_import_directory
+            else:
+                del sys._xoptions["snowflake_import_directory"]
+
+        with ExitStack() as stack:
+            stack.callback(cleanup_imports)
+
+            for module_path in self.imports:
```

Comment:
similar to my question above, in UDF implementation we have the `if udf_name in analyzer.session.udf._udf_level_imports:` check, but seems like we remove that concept in sp.

want to know why the diff

Reply:
`if udf_name in analyzer.session.udf._udf_level_imports:` is the same as `if sproc_name in self._sproc_level_imports:` below. UDF/Sproc level imports overrides session level imports. If no UDF/Sproc level imports specified, session level imports are used.